sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3304,6 +3304,15 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
         """.stripMargin).collect()
     }
   }
+
+  test("SPARK-29860: Fix dataType mismatch issue for InSubquery") {
+    withTable("ta", "tb") {
+      sql("create table ta(id Decimal(18,0)) using parquet")
+      sql("create table tb(id Decimal(19,0)) using parquet")
+      sql("select * from ta where id in (select id from tb)").collect()
+      sql("select * from tb where id in (select id from ta)").collect()
```

Comment:
Can you insert test data then check the answer with `assert`?

Reply:
thanks, I have updated the code and UT.